docs/vo_service_discovery/vo_service_discovery.rst

```diff
@@ -0,0 +1,183 @@
+The options include gamma-ray (wavelength below 0.1 Angstrom), x-ray
+(0.1 to 100 Angstrom), euv (100-1000 Angstrom), uv (1000 to 3000 Angstrom),
+optical (3000 to 10000 Angstrom), infrared (1 to 100 micron), millimeter (0.1 to
+10 mm), or radio (greater than 10 mm wavelength). Multiple bands can be specified
+with a comma-separated list.
+
+The ``source`` parameter is a string that is matched to any substring in ivoid.
+
+The ``publisher`` is the name of any publishing organization.
+
+``order_by`` can be specified to sort the result by any of the column names.
+
+The ``logic_string`` is any other string that is needed for the ADQL ``where``
+clause, and must start with the value ``' and '``.
+
+
+The ADQL query can be printed by specifying ``verbose=True``.
+
+.. code-block:: python
+
+    >>> result = Registry.query(service_type='tap', source='heasarc', verbose=True)
+
+    Registry:  sending query ADQL =
+              select res.waveband,res.short_name,cap.ivoid,res.res_description,
+              intf.access_url,res.reference_url,res_role.role_name as publisher,cap.cap_type as service_type
+              from rr.capability as cap
+                natural join rr.resource as res
+                natural join rr.interface as intf
+                natural join rr.res_role as res_role
+                 where cap.cap_type like '%tableaccess%' and cap.ivoid like '%heasarc%' and res_role.base_role = 'publisher'
+
+    Queried: https://vao.stsci.edu/RegTAP/TapService.aspx/sync
+
+    >>> print(result)
+
+    waveband short_name ...  publisher   service_type
+    -------- ---------- ... ------------ ------------
+                HEASARC ... NASA/HEASARC  tableaccess
+
+
+Counting available services
+===========================
+
+The `query_counts` method will enumerate how many services are available
+for each of the fields in the results table shown above.
+
+.. code-block:: python
+
+    >>> result = Registry.query_counts(field='service_type')
+    >>> print(result)
+
+           service_type       count_service_type
+    ------------------------- ------------------
+                   capability              59214
+                   conesearch              21453
+            simpleimageaccess                334
+         simplespectralaccess                115
+                  tableaccess                 97
+                      edition                 86
+                ceacapability                 56
+                      harvest                 41
+             simpletimeaccess                 31
+                  openskynode                 27
+                       search                  8
+    theoreticalspectralaccess                  5
+                     heliotap                  5
+          protospectralaccess                  4
+             simplelineaccess                  3
+
+We can get a list of the top 15 publishers:
+
+.. code-block:: python
+
+    >>> results = Registry.query_counts('publisher', 15, verbose=True)
```

Comment:
Recommend to specify the option `minimum`.

```suggestion
    >>> results = Registry.query_counts('publisher', minimum=15, verbose=True)
```

Reply:
Addressed.